### ISSUE TITLE
Including external python module dependencies fails

### ISSUE THREAD

**Issue by cogfor** [GoogleCloudPlatform/spark-on-k8s-operator#331]
TL;DR

I am using python with spark operator for 2.4.0 and GCS. Referencing external python modules using `pyFiles` results in a `ModuleNotFoundError`

**Intro**
My pyspark script needs some external (pypi) dependencies. So I did:

```pip install -t dependencies -r requirements.txt
cd dependencies
zip -r ../dependencies.zip .```

Then I created the SparkApplication yaml with these settings:

 ```deps:
    pyFiles:
      - "gs://regiolab-spark/dependencies.zip"
```

Note: I am using a custom `spark-py` docker image with gcs support

My python file (also on GCS) has:

```
python_dep_file_path = SparkFiles.get("requirements.zip")
conf = SparkConf() \
    .setAppName("GSDataReader") \
    .addPyFile(python_dep_file_path)
sc = SparkContext(conf=conf)
spark = SparkSession.builder.config(conf=sc.getConf()).getOrCreate()
```

**problem**
Still I am getting a: `ModuleNotFoundError: No module named 'geopy'`. A module that is in my dependencies.zip.

Note: I am using Spark Operator for Spark 2.4.0 installed by Helm with pull request #9048 with a custom docker image to add GCS support.

Is this a bug? Something I configured wrong?

SparkApplication Spec: https://pastebin.com/FKsqCTK3
Driver log: https://pastebin.com/2WKcG8BM

**Comment by charlesa101** [comment]
was this ever addressed, i am having the same issue

**Comment by pmesserschmidt1** [comment]
Was this problem solved? I'm facing the same problem here.
